### PR TITLE
fix(terminal): roll magnitude formatters into the next unit at rendered boundaries

### LINKED ISSUE #5312: [BUG] token/bytes magnitude formatters left behind by #5179 still render 1000.0k and 1024 B at unit boundaries
### Summary

Three magnitude formatters have the round-before-branch bug fixed in `format_token_count_compact` by #5179: they branch on the raw value but render rounded, so values just below a unit boundary display as `1000.0k` / `1024 B` / `1024.00 KiB` instead of rolling into the next unit.

### Expected vs actual behavior

**Expected:** values at a unit boundary roll up — `999_950` tokens → `1.0M`, `1023.5` bytes → `1.00 KiB`, `1_048_570.9` bytes → `1.00 MiB`.

**Actual:**

- `surfaces/shared/terminal/components/token_format.py` `format_token_count_short` (the spinner's `↓ tokens` counter, fed by cumulative `bytes_in // 4`) has no M tier at all and renders `999_950` → `1000.0k`, `2_500_000` → `2500.0k`.
- `platform/evidence/metric_summary.py` `_format_bytes` advances units while the **raw** amount ≥ 1024 but renders at `.0f`/`.2f`, so `1023.5` → `1024 B` and `1_048_570.9` → `1024.00 KiB` in evidence metric summaries.
- `surfaces/shared/terminal/agents/agents_view.py` `_format_tokens_per_min` rounds for the `<1000` comparison (a comment even explains why) but the k tier repeats the same bug one unit up: `999_950` → `1000.0k`, and there is no M tier.

### Steps to reproduce

From the repo root:

```
uv run python -c "
from surfaces.shared.terminal.components.token_format import format_token_count_short
for v in (999949, 999950, 2500000): print(v, '->', format_token_count_short(v))
from platform.evidence.metric_summary import _format_bytes
for v in (1023.5, 1048570.9): print(v, '->', _format_bytes(v))
"
```

Output:

```
999949 -> 999.9k
999950 -> 1000.0k
2500000 -> 2500.0k
1023.5 -> 1024 B
1048570.9 -> 1024.00 KiB
```

### Root cause / proposed fix

Same class as #5178/#5179: the branch compares the raw value against the boundary, but rendering rounds — so the window between "rounds to the boundary" and "reaches the boundary" falls through with a four-digit mantissa. Fix mirrors the merged #5179 pattern: branch on the rounded magnitude (`round(v / 1000, 1) < 1000`), add the missing M tier to the token formatters, and advance `_format_bytes`'s unit while the amount **rendered at that unit's precision** would still show 1024. Regression tests at each boundary. Happy to open a PR.

### Environment

- opensre @ current main (c1f671c02)
- Python via `uv run`, macOS

## Testing this from a surface

Run a turn long enough to push the token counter past 999,950 and the context
past a MiB boundary, then:

```
> /cost
> /context
> /fleet
```

**Pass:** `1.0M`, not `1000.0k`; `1.00 KiB`, not `1024 B`. The byte formatter is
also visible in the evidence metric summaries of an RCA:

```bash
uv run opensre investigate -i alert.json -o rca.json
grep -oE '[0-9.]+ (B|KiB|MiB)' rca.json | sort -u
```

### PR DESCRIPTION
**Pull request by cloudenochcsis**
Fixes #5312

#### Describe the changes you have made in this PR -

Three magnitude formatters repeated the round-before-branch bug fixed for `format_token_count_compact` in #5179: they branch on the raw value but render rounded, so values in the window between "rounds to the boundary" and "reaches the boundary" display with a four-digit mantissa instead of rolling into the next unit.

- `surfaces/shared/terminal/components/token_format.py` `format_token_count_short` — the spinner's `↓ tokens` counter, fed by cumulative `bytes_in // 4` — had no M tier at all: `999_950` → `1000.0k`, `2_500_000` → `2500.0k`.
- `platform/evidence/metric_summary.py` `_format_bytes` advanced units while the raw amount ≥ 1024 but renders at `.0f`/`.2f`: `1023.5` → `1024 B`, `1_048_570.9` → `1024.00 KiB`.
- `surfaces/shared/terminal/agents/agents_view.py` `_format_tokens_per_min` rounds for its `<1000` comparison (the comment above it explains exactly this bug) but repeated it one tier up: `999_950` → `1000.0k`.

All three now branch on the rounded magnitude per the merged #5179 pattern; the token formatters gain the missing M tier, and `_format_bytes` advances while the amount rendered at the unit's precision would still show 1024. Boundary regression tests added for each.

### Demo/Screenshot for feature changes and bug fixes -

Before (main):

```
999949 -> 999.9k
999950 -> 1000.0k
2500000 -> 2500.0k
1023.5 -> 1024 B
1048570.9 -> 1024.00 KiB
```

After (this branch):

```
999949 -> 999.9k
999950 -> 1.0M
2500000 -> 2.5M
1023.5 -> 1.00 KiB
1048570.9 -> 1.00 MiB
```

Focused tests + baseline checks (per CI.md):

```
uv run python -m pytest tests/interactive_shell/ui/test_streaming.py tests/tools/utils/test_metric_summary.py tests/shared/terminal/test_agents_view_integration.py -q   # 61 passed
make lint          # All checks passed!
make format-check  # 3618 files already formatted
make typecheck     # Success: no issues found in 1872 source files
```

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

I kept three local fixes rather than extracting a shared helper: the three formatters have deliberately different output contracts (`format_token_count_short` keeps a fixed `.1f`, `format_token_count_compact` strips trailing zeros, `_format_bytes` switches precision per unit, `_format_tokens_per_min` renders integers below 1k), so a shared function would need enough parameters to be harder to read than the three-line guard it replaces. `_format_bytes` checks `round(amount, precision) < 1024` with the precision the current unit actually renders at (`.0f` for B, `.2f` above), which is the exact condition under which the displayed string can reach "1024". The negative-value path in `_format_bytes` is covered by a test since the sign is re-applied after the unit walk.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

**Comment by cloudenochcsis** [comment]
@greptile review

File: infrastructure/evidence/metric_summary.py [hunk 1]
```diff
@@ -255,7 +255,12 @@ def _format_bytes(value: float) -> str:
...
     unit_index = 0
-    while amount >= 1024 and unit_index < len(units) - 1:
+    # Advance while the amount rendered at this unit's precision would still
+    # show 1024: 1023.5 rounds to "1024 B" at .0f and must roll to KiB.
+    while unit_index < len(units) - 1:
+        precision = 0 if unit_index == 0 else 2
+        if round(amount, precision) < 1024:
+            break
         amount /= 1024
         unit_index += 1
```
[the remark this answers is not in the document]
Comment (cloudenochcsis) [reply infrastructure/evidence/metric_summary.py:270]: Good catch — fixed in 55015d313. `round(nan, p) < 1024` is false at every precision, so the rendered-precision walk promoted NaN to "nan TiB" where the pre-change raw comparison left it at B. Added a NaN guard before the walk that keeps a magnitude-less value at the base unit (`nan B`, the previous rendering), with a regression test pinning it.

**Comment by cloudenochcsis** [comment]
@greptile review
